exoscale/api/compute.py

```diff
@@ -117,12 +95,15 @@ class ElasticIP(Resource):
     healthcheck_strikes_fail = attr.ib(default=None, repr=False)
 
     @classmethod
-    def _from_cs(cls, compute, res):
+    def _from_cs(cls, compute, res, zone=None):
```

Comment:
the `_from_cs()` classmethod now accepts an optional `zone` parameter, allowing the caller to provide an existing `Zone` object in order to save an API call during instantiation; if none provided, the classmethod resolves the zone with a `compute.get_zone()` lookup.